pkg/agent/openflow/client.go

```diff
@@ -236,6 +236,12 @@ type Client interface {
 	IsIPv4Enabled() bool
 	// Returns if IPv6 is supported on this Node or not.
 	IsIPv6Enabled() bool
+	// GenBasePacketOutBuilder generates a base IP packetOutBuilder, which can have more packet data added.
+	GenBasePacketOutBuilder(srcMAC string, dstMAC string, srcIP string, dstIP string, inPort uint32, outPort uint32) (binding.PacketOutBuilder, error)
```

Comment:
I don't know how others feel about this, but it doesn't seem right to me to have this method part of this interface, and have each operation require 2 separate method calls (e.g. `GenBasePacketOutBuilder` call followed by `SendTCPReject` to send a TCP reset packet).

It also contrasts with `SendTraceflowPacket`, and I am not sure why we want to have that discrepancy.

Reply:
I previously was kind of following how `SendTraceflowPacket` impl it. I used a single func `SendRejectPacket` to handle different protocol rejections. Since as I discussed with @jianjuns here https://github.com/vmware-tanzu/antrea/pull/1888#discussion_r582374024, I tried to separate it. Maybe I misunderstanding something. Any idea?

Reply:
My preference would be to avoid introducing new dependencies on the `pkg/ovs` package in this interface. I also prefer not to have helper methods in an interface and for me this qualifies as one. It's not really doing any work, it's just encapsulating some header values into an object that can then be consumed by the method doing the actual work. However, if this design works for @jianjuns, it's good enough for me. It can always be changed in the future.

Reply:
I have the same feeling that `GenBasePacketOutBuilder` shouldn't be an interface of openflow client which is used to interact with OVS via openflow protocol.

Reply:
I have submitted another version. How do you guys think about this one?